src/operator/tensor/elemwise_binary_op.h

```diff
@@ -32,6 +34,105 @@ void BinaryCompute(const nnvm::NodeAttrs& attrs,
   });
 }
 
+// TODO make use of templated OP
+template<typename xpu, typename OP>
+void BinaryComputeNDSpSp(const nnvm::NodeAttrs& attrs,
+                         const OpContext& ctx,
+                         const std::vector<NDArray>& inputs,
+                         const std::vector<OpReqType>& req,
+                         const std::vector<NDArray>& outputs) {
+  std::cout << "BinaryComputeNDSpSp invoked\n";
+  CHECK(inputs.size() == 2);
+  CHECK(outputs.size() == 1);
+  auto &nd_l = inputs[0];
+  auto &nd_r = inputs[1];
+  auto &output = outputs[0];
+
+  // TODO support other sparse types, too
+  CHECK(nd_l.chunk_type() == kRowSparseChunk);
+  // Memory Estimation
+  auto num_rows_l = nd_l.aux_shape(0)[0];
+  auto num_rows_r = nd_r.aux_shape(0)[0];
+  // This is (roughly) the number of result rows
+  output.CheckAndAlloc({TShape({num_rows_l + num_rows_r})});
+
+  // Indices
+  mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
+  auto indices_l = nd_l.aux_data(0).FlatTo1D<xpu, ROW_SPARSE_TYPE>(s);
+  auto indices_r = nd_r.aux_data(0).FlatTo1D<xpu, ROW_SPARSE_TYPE>(s);
+  auto indices_out = output.aux_data(0).FlatTo1D<xpu, ROW_SPARSE_TYPE>(s);
```

Comment:
A general comment is that since this implements row sparse format of tensors, can we have meaningful interfaces to get the number of rows and row index array from a NDArray? Currently, these two attributes are retrieved from `NDArray::aux_shape(0)[0]` and `NDArray::aux_data(0).FlatTo1D<xpu, ROW_SPARSE_TYPE>(s)`, which seems not very natural. I would have to look into the detailed implementation of Chunk::CheckAndAlloc to understand the meaning of these.